libs/tests/unit/test_litmus_auth_interface.py

```diff
@@ -0,0 +1,205 @@
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+import dataclasses
+import json
+
+import pytest
+from ops import CharmBase
+from ops.testing import Context, State
+
+from litmus_libs.interfaces import (
+    Endpoint,
+    LitmusAuthDataProvider,
+    LitmusAuthDataRequirer,
+)
+
+
+@pytest.mark.parametrize(
+    "input, expected",
+    (
+        (
+            Endpoint(
+                grpc_server_host="host",
+                grpc_server_port=80,
+            ),
+            {
+                "grpc_server_host": json.dumps("host"),
+                "grpc_server_port": json.dumps(80),
+                "insecure": json.dumps(False),
+                "version": json.dumps(0),
+            },
+        ),
+        (
+            Endpoint(
+                grpc_server_host="host",
+                grpc_server_port=80,
+                insecure=True,
+            ),
+            {
+                "grpc_server_host": json.dumps("host"),
+                "grpc_server_port": json.dumps(80),
+                "insecure": json.dumps(True),
+                "version": json.dumps(0),
+            },
+        ),
+    ),
+)
+def test_provider_publish_endpoint(litmus_auth, input, expected):
+    # GIVEN a charm that provides litmus-auth
+    ctx = Context(
+        CharmBase,
+        meta={"name": "provider", "provides": {"litmus-auth": {"interface": "litmus_auth"}}},
+    )
+    with ctx(
+        state=State(
+            relations={litmus_auth},
+            leader=True,
+        ),
+        event=ctx.on.update_status(),
+    ) as mgr:
+        charm = mgr.charm
```

Comment:
overall I don't mind this style of testing with a throwaway charmbase context, but I think we should have at least one 'proper' API test to verify that the usage instructions we're documenting in the requirer/provider docstrings really works as intended.